File: abc097_b.kt

fun main(args:Array<String>) {
  val b = readLine()!!.toDouble()
  val c = (2..999).map {
    val ba = it.toDouble()
    val bb = Math.pow(b, 1/ba).toInt().toDouble()
    Math.pow(bb.toDouble(), ba).toInt()
  }.toList().max()!!
  println( c )
}

File: abc097_b.py
import math
import itertools
X=int(input())

if X==1:
    print(1)
    exit()

ans = 0 
for x in reversed(range(2,X)):
    for p in itertools.count(2):
        if pow(x,p) <= X:
            ans = max(ans, pow(x,p))
        else:
            break
print(ans)
